wordpress-mobile/WordPress-iOS
# Like Notifications: Show error when fetching Likes fails or there is no connection.

Ref #15662 

An error view is shown when:
- There are no cached Likes and there is no internet connection.
- Fetching Likes fails.

To test:

---
Prereqs:
- Start with a fresh install so you have no cached Likes.
- Enable `newLikeNotifications` feature flag.

---
No connection:
- Go to Notifications > Likes.
- Disable internet connection.
- Select a notification.
- Verify the `Unable to load this content` view is displayed.

---
Fetch error:
- Re-enable internet connection.
- You may want to manually cause fetching to fail for Comments and/or Posts. For example in `CommentService+Likes` / `PostService+Likes`, change `getLikesFor` to just call `failure`:
```
        remote.getLikesForCommentID(commentID) { remoteLikeUsers in
            
            failure(nil)
            
//            self.createNewUsers(from: remoteLikeUsers, commentID: commentID, siteID: siteID) {
//                let users = self.likeUsersFor(commentID: commentID, siteID: siteID)
//                success(users)
//                LikeUserHelper.purgeStaleLikes()
//            }
```
- Go to Notifications > Likes.
- Select a notification.
- Verify the `Oops` view is displayed.

---
`All` notifications:
- With either of the error views displayed for a Like Notification, select the `All` filter.
- Select any notification.
- Go through notifications by tapping the next and previous buttons on the nav bar.
- Verify the notification details are displayed correctly. That is, the error view is only displayed for the Like Notification and other notifications display their content as expected.

---
| No Connection | Fetch Fail |
|--------|-------|
| ![Simulator Screen Shot - iPhone 12 Pro Max - 2021-05-13 at 16 18 20](https://user-images.githubusercontent.com/1816888/118195427-74f6c700-b408-11eb-89f2-4f92d98d28c4.png) | ![Simulator Screen Shot - iPhone 12 Pro Max - 2021-05-13 at 16 26 17](https://user-images.githubusercontent.com/1816888/118195447-7d4f0200-b408-11eb-93ea-777e7952c222.png) |

---
## Regression Notes
1. Potential unintended areas of impact
N/A. Feature incomplete.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature incomplete.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature incomplete.

PR submission checklist:
